pkgs/build-support/kernel/make-initrd.nix

```diff
@@ -33,7 +33,10 @@ stdenv.mkDerivation rec {
   # Note: we don't use closureInfo yet, as that won't build with nix-1.x.
   # See #36268.
   exportReferencesGraph =
-    map (x: [("closure-" + baseNameOf x.symlink) x.object]) contents;
+    lib.zipListsWith 
+      (x: i: [("closure-${toString i}") x.object]) 
+      contents 
+      (lib.range 0 (lib.length contents - 1));
```

Comment:
Another variant - remove only forbidden symbols
```
with builtins; lib.concatStringsSep "-" (filter (x: !(isList x)) 
    (split "[^a-zA-Z0-9_=.?-]+" "autovt@.service"))
```
Allowed char list is based on https://github.com/NixOS/nix/blob/7cc1a2593ebece4df28abc86c8b83f5d14dc23a0/src/libstore/store-api.cc#L85-L101

Reply:
that would potentially cause collisions.  the enumerator makes sure each name is unique

Reply:
ok, maybe then include both - censored name + int number? As of now I think this can be merged, however I don't know, maybe human-readable name was used for debugging.

Reply:
@edolstra do you have any strong opinions on readable names? I seem to be the first to touch this part of the code in 11 years :P

Reply:
@danbst I have added the name escaping code as you suggested